protocol.md

```diff
@@ -870,6 +902,12 @@ export interface TextDocumentClientCapabilities {
 ```typescript
 interface ClientCapabilities {
 	/**
+	 * The client supports receiving the result solely through $/partialResult notifications for requests from the client to the server.
+	 * If true, the server should send all data through $/partialResult and send null as the result in the final response.
```

Comment:
What exactly doesn't feel right about this? The second sentence is completely intentional. Do you think we should rather send the full result again? That would be unnecessary overhead.

 > I worry about jsonrpc2 implementations making it difficult to construct a consistent order of when it receives a notification and a response, since we are now dictating an ordering requirement on responses/notifications that never existed before.

True, that requirement is part of the streaming support - not related to sending `null` as the final result. I don't know why an implementation should mess up the ordering though - why would it emit a certain message after another even if was received first?